sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/EmptyLineRegexCheck.kt

```diff
@@ -0,0 +1,258 @@
+
+        return when (preParent) {
+            is KtExpression -> if (preParent.getResolvedCall(bindingContext) matches PATTERN_FIND) extractArgument(resolvedCall) else null
+            is KtProperty -> if (preParent.usages().any { it.getParentResolvedCall(bindingContext) matches PATTERN_FIND }) {
+                extractArgument(resolvedCall)
+            } else null
+            else -> null
+        }
+    }
+
+    private fun getStringInRegexFind(ref: KtExpression, bindingContext: BindingContext): KtExpression? {
+        val resolvedCall = (ref.parent as? KtExpression)?.getResolvedCall(bindingContext) ?: return null
+        return if (resolvedCall matches REGEX_FIND) extractArgument(resolvedCall) else null
+    }
+    
+    private fun extractArgument(resolvedCall: ResolvedCall<out CallableDescriptor>): KtExpression? {
+        val arguments = resolvedCall.valueArgumentsByIndex
+        return if (arguments == null
+            || arguments.isEmpty()
+            || arguments[0] == null
+            || arguments[0] !is ExpressionValueArgument
+        ) null
+        else (arguments[0] as ExpressionValueArgument).valueArgument?.getArgumentExpression()
+    }
+
+    private fun KtExpression.canBeEmpty(bindingContext: BindingContext): Boolean =
+        when (val deparenthesized = this.deparenthesize()) {
+            is KtNameReferenceExpression -> {
+                val runtimeStringValue = deparenthesized.predictRuntimeStringValue(bindingContext)
+                runtimeStringValue?.isEmpty()
+                    ?: deparenthesized.findAllUsages {
+                        (it.parent as? KtExpression)?.getResolvedCall(bindingContext) matches STRING_IS_EMPTY
+                    }.isEmpty()
+            }
+            else ->
+                predictRuntimeStringValue(bindingContext)?.isEmpty() ?: false
+        }
+
+
+    private fun checkRegexInReplace(callExpression: KtCallExpression, kotlinFileContext: KotlinFileContext, parent: PsiElement) {
+        val bindingContext = kotlinFileContext.bindingContext
+        when (parent) {
+            is KtProperty -> {
+                parent.usages()
+                    .mapNotNull { it.getParentResolvedCall(bindingContext) }
+                    .filter { it matches STRING_REPLACE && it.getReceiverExpression()?.canBeEmpty(bindingContext) == true }
+                    .firstOrNull()
+                    ?.let { kotlinFileContext.reportIssue(it.getFirstArgumentExpression()!!, MESSAGE, listOf(kotlinFileContext.secondaryOf(callExpression))) }
+            }
+            else -> {
+                val resolvedCall = callExpression.getParentResolvedCall(bindingContext)
+                if (resolvedCall matches STRING_REPLACE
+                    && resolvedCall?.getReceiverExpression()?.canBeEmpty(bindingContext) == true) {
+                    kotlinFileContext.reportIssue(resolvedCall.getFirstArgumentExpression()!!, MESSAGE)                    
```

Comment:
This looks a bit confusing on first sight - can you either put the condition on one line or the `) {` on a separate line?